packages/lesswrong/server/serverSentEvents.ts

```diff
@@ -71,10 +75,137 @@ export function addServerSentEventsEndpoint(app: Express) {
+    return;
+  }
+
+  // TODO actually load real typingIndicators from the db
+
+  // const newTypingIndicators = await TypingIndicators.find({
+  //   lastUpdated: {$gt: lastNotificationCheck}
+  // }, {
+  //   projection: {userId:1, lastUpdated:1, documentId:1}
+  // }).fetch();
+
+  const newTypingIndicatorsDummy: TypingIndicatorInfo[] = [{
+      _id: "1",
+      userId: "raemon r38pkCm7wF4M44MDQ", // raemon
+      documentId: "5qYcLQ8vcXjcpeC2D", // dialogue on stuff
+      lastUpdated: new Date()
+    }, {
+      _id: "2",
+      userId: "jacob gXeEWGjTWyqgrQTzR", // jacobjacob
+      documentId: "5qYcLQ8vcXjcpeC2D", // dialogue on stuff
+      lastUpdated: new Date()
+    },
+    {
+      _id: "3",
+      userId: "jacobjacob gXeEWGjTWyqgrQTzR", // jacobjacob
+      documentId: "cCnK8CxNKZqxakmsG", // coconut
+      lastUpdated: new Date()
+    },
+    {
+      _id: "4",
+      userId: "testcob ultron i7cWXn8ApqsBaQDre", // testcob ultron
+      documentId: "5qYcLQ8vcXjcpeC2D", // dialogue on stuff
+      lastUpdated: new Date()
+    },
+    {
+      _id: "5",
+      userId: "raemontest kyf5Dfouz6c2udioL", // raemon test
+      documentId: "cCnK8CxNKZqxakmsG", // coconut
+      lastUpdated: new Date()
+    },
+    {
+      _id: "6",
+      userId: "habryka XtphY3uYHwruKqDyG", // habryka
+      documentId: "Bc9DHPrFJAGZ26bum", // otter world (includes jacobjacob, kave, habryka, raemo)
+      lastUpdated: new Date()
+    },
+    {
+      _id: "7",
+      userId: "kave 55XxDBpfKkkBPm9H8", // kave
+      documentId: "Bc9DHPrFJAGZ26bum", // otter world (includes jacobjacob, kave, habryka, raemon)
+      lastUpdated: new Date()
+    }
+  ]
+
+  //dialog on stuff has 
+    // raemon, testcob ultron
+  // coconut has
+    // jacob, raemontest
+  // Otter world
+    // jacobjacob, kave, habryka, raemon
+
+  const listOfPosts = uniq(newTypingIndicatorsDummy.map(indicator => indicator.documentId))
+
+  const posts = await Posts.find({
+    _id: {$in:listOfPosts}}, {
+    projection: {_id: 1, userId: 1, coauthorStatuses: 1, shareWithUsers: 1}
+  }).fetch()
+
+  const postsWithUserIds: Record<string, string[]> = {}
+  for (let post of posts) {
+    const coauthorIds = post.coauthorStatuses ? post.coauthorStatuses.map(coauthor => coauthor.userId) : []
+    const shareWithUsersIds = post.shareWithUsers ? post.shareWithUsers : []
+    postsWithUserIds[post._id] = [...coauthorIds, ...shareWithUsersIds, post.userId]
+  }
+  
+  const usersWithPostIds: Record<string, string[]> = {}
+  // map users to posts
+  for (let postId of Object.keys(postsWithUserIds)) {
+    const userIds = postsWithUserIds[postId]
+    for (let userId of userIds) {
+      if (userId in usersWithPostIds) {
+        usersWithPostIds[userId].push(postId)
+      } else {
+        usersWithPostIds[userId] = [postId]
+      }
+    }
+  }
+
+  const postsWithTypingIndicators: Record<string, TypingIndicatorInfo[]> = {}
+  for (let typingIndicator of newTypingIndicatorsDummy) {
```

Comment:
Don't forget to filter these for `collectionName: 'Posts'` (assuming the non-post ones, should they ever exist, will be supported elsewhere)

Reply:
ignoring per conversatoin in person